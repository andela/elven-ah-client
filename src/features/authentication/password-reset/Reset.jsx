import React from 'react';
import { PropTypes } from 'prop-types';

/* eslint-disable react/no-array-index-key */
const Reset = ({
  handleChange, values, handleReset, errors,
}) => (
  <div className="mx-auto centered text-center col-xl-4 col-lg-4 col-md-6 col-sm-6 col-xs-6">
    <form className="form-login" onSubmit={handleReset}>
      <h4>Enter your new password</h4>
      <div className="form-label-group mb-3">
        <input
          onChange={handleChange}
          type="password"
          className="form-control form-input"
          id="password"
          value={values.password}
          placeholder="Enter your password"
          required
        />
        {errors.password ? errors.password.map((error, index) => <span className="invalid-feedback" key={index}>{error}</span>) : ''}
      </div>
      <div className="form-label-group mb-3">
        <input
          onChange={handleChange}
          type="password"
          className="form-control form-input"
          id="confirmPassword"
          value={values.confirmPassword}
          placeholder="Repeat password"
          required
        />
        <span className="invalid-feedback">{!errors.password && !errors.confirmPassword ? errors.message : ''}</span>
        {errors.confirmPassword ? errors.confirmPassword.map((error, index) => <span className="invalid-feedback" key={index}>{error}</span>) : ''}
      </div>
      <button className="btn mx-auto btn-block auth-submit-btn" type="submit"> Update Password</button>
    </form>
  </div>
);

Reset.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  errors: PropTypes.shape({}).isRequired,
  values: PropTypes.shape({}).isRequired,
};

export default Reset;
