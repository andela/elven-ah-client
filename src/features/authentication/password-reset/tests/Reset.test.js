import React from 'react';
import { shallow } from 'enzyme';

import Reset from '../Reset';

const mockFunction = jest.fn();
const errors = {
  password: ['The password field is required'],
  confirmPassword: ['The password field is required'],
};
const values = {
  email: '',
  confirmPassword: '',
  password: '',
};

describe('Test the Reset password presentation component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Reset
      handleChange={mockFunction}
      handleReset={mockFunction}
      errors={errors}
      values={values}
    />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without errors', () => {
    const wrapper = shallow(<Reset
      handleChange={mockFunction}
      handleReset={mockFunction}
      errors={errors}
      values={values}
    />);
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(2);
    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper.find('span.invalid-feedback')).toHaveLength(3);
  });
});
