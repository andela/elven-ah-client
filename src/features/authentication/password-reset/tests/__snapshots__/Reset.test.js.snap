// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Reset password presentation component renders without crashing 1`] = `
<div
  className="mx-auto centered text-center col-xl-4 col-lg-4 col-md-6 col-sm-6 col-xs-6"
>
  <form
    className="form-login"
    onSubmit={[Function]}
  >
    <h4>
      Enter your new password
    </h4>
    <div
      className="form-label-group mb-3"
    >
      <input
        className="form-control form-input"
        id="password"
        onChange={[Function]}
        placeholder="Enter your password"
        required={true}
        type="password"
        value=""
      />
      <span
        className="invalid-feedback"
        key="0"
      >
        The password field is required
      </span>
    </div>
    <div
      className="form-label-group mb-3"
    >
      <input
        className="form-control form-input"
        id="confirmPassword"
        onChange={[Function]}
        placeholder="Repeat password"
        required={true}
        type="password"
        value=""
      />
      <span
        className="invalid-feedback"
      />
      <span
        className="invalid-feedback"
        key="0"
      >
        The password field is required
      </span>
    </div>
    <button
      className="btn mx-auto btn-block auth-submit-btn"
      type="submit"
    >
       Update Password
    </button>
  </form>
</div>
`;
